jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -50,6 +56,8 @@ def specviz_spectrum1d_parser(app, data, data_label=None, format=None, show_in_v
             except IORegistryError:
                 # Multi-extension files may throw a registry error
                 data = SpectrumList.read(str(path), format=format)
+        elif path.is_dir():
+            data = SpectrumList.read(str(path), format=format)
```

Comment:
Shouldn't there be some checks here to make sure the files in the directory are valid?  What does `SpectrumList.read` do here when a directory of junk files are passed in?

Reply:
Also, what kind of directory should this expect?  It looks like a flattened directory?

Reply:
The checks are part of the `SpectrumList.read` functionality. I can't find in the code how the path is expanded (`glob`? @rosteen ), but I think typically it will be a flattened directory. Typical use would be to plot together the several `x1d` pieces associated to a MIRI LRS spectral data set, as done in the notebook.

Reply:
Judging by the concept notebook and the testing I've done, it looks like this expects a flat directory (or anyway just reads any matching files in the directory level given), in the case of the concept notebook reading in and combining only the x1d files there (I believe the MIRI MRS specutils reader is doing the work, since iirc that was the specific case for which this was requested).

I think there is a check needed - I tried this on a directory with two SDSS spectra in it, and `SpectrumList.read()` returns an empty list in that case since the SDSS reader isn't set up to handle the directory case with `SpectrumList`. I can add a warning in that case that the directory-read functionality is currently JWST specific, and add a bit of documentation about what will happen if you load in a pre-created `SpectrumList`.

Reply:
I'll note that the `specutils` [docs on this](https://specutils.readthedocs.io/en/stable/custom_loading.html#loading-multiple-spectra) are pretty sparse, and mostly consist of fairly opaque example code from the JWST readers.

Reply:
I wonder if this works with level 2 JWST x1d files, but my guess is the loaders only work on level 3 data.  I never checked that, just always assumed it only works on L3.  I'll also add that this functionality will be useful for MAST as well, but files on disk for MAST have a structured directory system rather than flattened.   That can be a future work item.